src/main/java/com/google/firebase/messaging/TopicManagementResponse.java

```diff
@@ -0,0 +1,124 @@
+/*
+ * Copyright 2017 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.firebase.messaging;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import com.google.firebase.internal.NonNull;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * The response produced by FCM topic management operations.
+ */
+public class TopicManagementResponse {
+
+  private static final String UNKNOWN_ERROR = "unknown-error";
+
+  // Server error codes as defined in https://developers.google.com/instance-id/reference/server
+  // TODO: Should we handle other error codes here (e.g. PERMISSION_DENIED)?
+  private static final Map<String, String> ERROR_CODES = ImmutableMap.<String, String>builder()
```

Comment:
Can we move these to a single place so the Admin SDK can share these errors?

Reply:
The backend error codes (keys of the map) are specific to the services we invoke at various places. I'm not sure there's a lot to be gained by sharing them across the package/SDK. What sort of sharing were you thinking of? Put them all in a shared `FirebaseMessagingConstants` class?